python/llm/src/ipex_llm/transformers/models/llama.py

```diff
@@ -312,7 +312,10 @@ def should_use_xetla_mm_qkv(self, device):
     full_attn = self.q_proj.out_len == self.k_proj.out_len == self.v_proj.out_len
     supported_qtype = self.q_proj.qtype == SYM_INT4 and full_attn
     supported_qtype = supported_qtype or self.q_proj.qtype == FP8E5
-    enable_xetla = self.q_proj.enable_xetla
+    if self.q_proj.qtype == SYM_INT4 or self.q_proj.qtype == FP8E5:
+        enable_xetla = self.q_proj.enable_xetla
+    else:
```

Comment:
Are you sure other dtypes all do not need xetla?
maybe change `else` to `elif BF16`

Reply:
Here is the supported qtype of xetla.
```
supported_qtype = self.q_proj.qtype == SYM_INT4 and full_attn
supported_qtype = supported_qtype or self.q_proj.qtype == FP8E5
```
On XPU, the method returns True when q_proj.qtype is among supported_qtypes(SYM_INT4/FP8E5) and enable_xetla is True.
https://github.com/intel-analytics/ipex-llm/blob/0d518aab8d69ccf6d15f247fdf73fed8c1f45e49/python/llm/src/ipex_llm/transformers/models/llama.py#L316
So if qtype is other types, just set enable_xetla to False.

Reply:
> Are you sure other dtypes all do not need xetla? maybe change `else` to `elif BF16`

Here is the supported qtype of xetla.
```
supported_qtype = self.q_proj.qtype == SYM_INT4 and full_attn
supported_qtype = supported_qtype or self.q_proj.qtype == FP8E5
```
On XPU, the method returns True when q_proj.qtype is among supported_qtypes(SYM_INT4/FP8E5) and enable_xetla is True.
https://github.com/intel-analytics/ipex-llm/blob/0d518aab8d69ccf6d15f247fdf73fed8c1f45e49/python/llm/src/ipex_llm/transformers/models/llama.py#L316
So if qtype is other types, just set enable_xetla to False.